### ISSUE TITLE
Unhandled exception executing command 'f claim' in f: Argument 6 passed to FactionsPro\FactionMain::drawPlot() must be an instance of FactionsPro

### ISSUE THREAD

**Issue by teracube** [poggit-orphanage/FactionsPro_1.4#7]
PocketMine-MP 1.7dev-372
FactionsPro 1.4.5-7
https://poggit.pmmp.io/r/15821/FactionsPro_1.4_dev-25.phar

```2017-11-06 [16:38:45] [Server thread/CRITICAL]: Unhandled exception executing command 'f claim' in f: Argument 6 passed to FactionsPro\FactionMain::drawPlot() must be an instance of FactionsPro\Level, instance of pocketmine\level\Level given, called in phar:///home/gsp_1485/g3.hmserv.eu-19144/plugins/FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands.php on line 340
2017-11-06 [16:38:45] [Server thread/CRITICAL]: TypeError: "Argument 6 passed to FactionsPro\FactionMain::drawPlot() must be an instance of FactionsPro\Level, instance of pocketmine\level\Level given, called in phar:///home/gsp_1485/g3.hmserv.eu-19144/plugins/FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands.php on line 340" (EXCEPTION) in "FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionMain" at line 192
2017-11-06 [16:38:45] [Server thread/DEBUG]: #0 FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands(340): FactionsPro\FactionMain->drawPlot(pocketmine\Player object, string CrystalliaPvP, double -80, double 1, double -230, pocketmine\level\Level object, integer 25)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #1 FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionMain(93): FactionsPro\FactionCommands->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2017-11-06 [16:38:45] [Server thread/DEBUG]: #2 src/pocketmine/command/PluginCommand(58): FactionsPro\FactionMain->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2017-11-06 [16:38:45] [Server thread/DEBUG]: #3 src/pocketmine/command/SimpleCommandMap(258): pocketmine\command\PluginCommand->execute(pocketmine\Player object, string f, array Array())
2017-11-06 [16:38:45] [Server thread/DEBUG]: #4 src/pocketmine/Server(1979): pocketmine\command\SimpleCommandMap->dispatch(pocketmine\Player object, string f claim)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #5 src/pocketmine/Player(2141): pocketmine\Server->dispatchCommand(pocketmine\Player object, string f claim)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #6 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(220): pocketmine\Player->chat(string /f claim)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #7 src/pocketmine/network/mcpe/protocol/CommandRequestPacket(47): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleCommandRequest(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #8 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\CommandRequestPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #9 src/pocketmine/network/mcpe/protocol/BatchPacket(115): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #10 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #11 src/pocketmine/Player(3025): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #12 src/pocketmine/network/mcpe/RakLibInterface(143): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #13 src/raklib/server/ServerHandler(99): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 83.79.238.53:55040, raklib\protocol\EncapsulatedPacket object, integer 0)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #14 src/pocketmine/network/mcpe/RakLibInterface(82): raklib\server\ServerHandler->handlePacket()
2017-11-06 [16:38:45] [Server thread/DEBUG]: #15 src/pocketmine/network/Network(89): pocketmine\network\mcpe\RakLibInterface->process()
2017-11-06 [16:38:45] [Server thread/DEBUG]: #16 src/pocketmine/Server(2513): pocketmine\network\Network->processInterfaces()
2017-11-06 [16:38:45] [Server thread/DEBUG]: #17 src/pocketmine/Server(2269): pocketmine\Server->tick()
2017-11-06 [16:38:45] [Server thread/DEBUG]: #18 src/pocketmine/Server(2144): pocketmine\Server->tickProcessor()
2017-11-06 [16:38:45] [Server thread/DEBUG]: #19 src/pocketmine/Server(1734): pocketmine\Server->start()
2017-11-06 [16:38:45] [Server thread/DEBUG]: #20 src/pocketmine/PocketMine(564): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/, string /home/gsp_1485/g3.hmserv.eu-19144/, string /home/gsp_1485/g3.hmserv.eu-19144/plugins/)
2017-11-06 [16:38:45] [Server thread/DEBUG]: #21 /home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar(1): require(string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
2017-11-06 [16:38:51] [Server thread/CRITICAL]: Unhandled exception executing command 'f claim' in f: Argument 6 passed to FactionsPro\FactionMain::drawPlot() must be an instance of FactionsPro\Level, instance of pocketmine\level\Level given, called in phar:///home/gsp_1485/g3.hmserv.eu-19144/plugins/FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands.php on line 340
2017-11-06 [16:38:51] [Server thread/CRITICAL]: TypeError: "Argument 6 passed to FactionsPro\FactionMain::drawPlot() must be an instance of FactionsPro\Level, instance of pocketmine\level\Level given, called in phar:///home/gsp_1485/g3.hmserv.eu-19144/plugins/FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands.php on line 340" (EXCEPTION) in "FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionMain" at line 192
2017-11-06 [16:38:51] [Server thread/DEBUG]: #0 FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands(340): FactionsPro\FactionMain->drawPlot(pocketmine\Player object, string CrystalliaPvP, double -80, double 1, double -230, pocketmine\level\Level object, integer 25)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #1 FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionMain(93): FactionsPro\FactionCommands->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2017-11-06 [16:38:51] [Server thread/DEBUG]: #2 src/pocketmine/command/PluginCommand(58): FactionsPro\FactionMain->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2017-11-06 [16:38:51] [Server thread/DEBUG]: #3 src/pocketmine/command/SimpleCommandMap(258): pocketmine\command\PluginCommand->execute(pocketmine\Player object, string f, array Array())
2017-11-06 [16:38:51] [Server thread/DEBUG]: #4 src/pocketmine/Server(1979): pocketmine\command\SimpleCommandMap->dispatch(pocketmine\Player object, string f claim)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #5 src/pocketmine/Player(2141): pocketmine\Server->dispatchCommand(pocketmine\Player object, string f claim)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #6 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(220): pocketmine\Player->chat(string /f claim)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #7 src/pocketmine/network/mcpe/protocol/CommandRequestPacket(47): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleCommandRequest(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #8 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\CommandRequestPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #9 src/pocketmine/network/mcpe/protocol/BatchPacket(115): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #10 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #11 src/pocketmine/Player(3025): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #12 src/pocketmine/network/mcpe/RakLibInterface(143): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #13 src/raklib/server/ServerHandler(99): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 83.79.238.53:55040, raklib\protocol\EncapsulatedPacket object, integer 0)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #14 src/pocketmine/network/mcpe/RakLibInterface(84): raklib\server\ServerHandler->handlePacket()
2017-11-06 [16:38:51] [Server thread/DEBUG]: #15 src/pocketmine/network/Network(89): pocketmine\network\mcpe\RakLibInterface->process()
2017-11-06 [16:38:51] [Server thread/DEBUG]: #16 src/pocketmine/Server(2513): pocketmine\network\Network->processInterfaces()
2017-11-06 [16:38:51] [Server thread/DEBUG]: #17 src/pocketmine/Server(2269): pocketmine\Server->tick()
2017-11-06 [16:38:51] [Server thread/DEBUG]: #18 src/pocketmine/Server(2144): pocketmine\Server->tickProcessor()
2017-11-06 [16:38:51] [Server thread/DEBUG]: #19 src/pocketmine/Server(1734): pocketmine\Server->start()
2017-11-06 [16:38:51] [Server thread/DEBUG]: #20 src/pocketmine/PocketMine(564): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/, string /home/gsp_1485/g3.hmserv.eu-19144/, string /home/gsp_1485/g3.hmserv.eu-19144/plugins/)
2017-11-06 [16:38:51] [Server thread/DEBUG]: #21 /home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar(1): require(string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
2017-11-06 [16:39:00] [Server thread/CRITICAL]: Unhandled exception executing command 'f claim' in f: Argument 6 passed to FactionsPro\FactionMain::drawPlot() must be an instance of FactionsPro\Level, instance of pocketmine\level\Level given, called in phar:///home/gsp_1485/g3.hmserv.eu-19144/plugins/FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands.php on line 340
2017-11-06 [16:39:00] [Server thread/CRITICAL]: TypeError: "Argument 6 passed to FactionsPro\FactionMain::drawPlot() must be an instance of FactionsPro\Level, instance of pocketmine\level\Level given, called in phar:///home/gsp_1485/g3.hmserv.eu-19144/plugins/FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands.php on line 340" (EXCEPTION) in "FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionMain" at line 192
2017-11-06 [16:39:00] [Server thread/DEBUG]: #0 FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionCommands(340): FactionsPro\FactionMain->drawPlot(pocketmine\Player object, string CrystalliaPvP, double -72, double 1, double -233, pocketmine\level\Level object, integer 25)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #1 FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionMain(93): FactionsPro\FactionCommands->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2017-11-06 [16:39:00] [Server thread/DEBUG]: #2 src/pocketmine/command/PluginCommand(58): FactionsPro\FactionMain->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2017-11-06 [16:39:00] [Server thread/DEBUG]: #3 src/pocketmine/command/SimpleCommandMap(258): pocketmine\command\PluginCommand->execute(pocketmine\Player object, string f, array Array())
2017-11-06 [16:39:00] [Server thread/DEBUG]: #4 src/pocketmine/Server(1979): pocketmine\command\SimpleCommandMap->dispatch(pocketmine\Player object, string f claim)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #5 src/pocketmine/Player(2141): pocketmine\Server->dispatchCommand(pocketmine\Player object, string f claim)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #6 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(220): pocketmine\Player->chat(string /f claim)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #7 src/pocketmine/network/mcpe/protocol/CommandRequestPacket(47): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleCommandRequest(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #8 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\CommandRequestPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #9 src/pocketmine/network/mcpe/protocol/BatchPacket(115): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #10 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #11 src/pocketmine/Player(3025): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #12 src/pocketmine/network/mcpe/RakLibInterface(143): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #13 src/raklib/server/ServerHandler(99): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 83.79.238.53:55040, raklib\protocol\EncapsulatedPacket object, integer 0)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #14 src/pocketmine/network/mcpe/RakLibInterface(82): raklib\server\ServerHandler->handlePacket()
2017-11-06 [16:39:00] [Server thread/DEBUG]: #15 src/pocketmine/network/Network(89): pocketmine\network\mcpe\RakLibInterface->process()
2017-11-06 [16:39:00] [Server thread/DEBUG]: #16 src/pocketmine/Server(2513): pocketmine\network\Network->processInterfaces()
2017-11-06 [16:39:00] [Server thread/DEBUG]: #17 src/pocketmine/Server(2269): pocketmine\Server->tick()
2017-11-06 [16:39:00] [Server thread/DEBUG]: #18 src/pocketmine/Server(2144): pocketmine\Server->tickProcessor()
2017-11-06 [16:39:00] [Server thread/DEBUG]: #19 src/pocketmine/Server(1734): pocketmine\Server->start()
2017-11-06 [16:39:00] [Server thread/DEBUG]: #20 src/pocketmine/PocketMine(564): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/, string /home/gsp_1485/g3.hmserv.eu-19144/, string /home/gsp_1485/g3.hmserv.eu-19144/plugins/)
2017-11-06 [16:39:00] [Server thread/DEBUG]: #21 /home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar(1): require(string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/src/pocketmine/PocketMine.php)```

**Comment by Awzaw** [comment]
I wish I didn't have to maintain 2 slightly different versions of this plugin! Please try again, and note that I didn't bump the version this time.

**Comment by teracube** [comment]
@Awzaw 
thanks for your job
With latest build dev 26 :

this issue seem to be solved : 
 TypeError: "Argument 6 passed .... on line 340" (EXCEPTION) in "FactionsPro_1.4_dev-25.phar/src/FactionsPro/FactionMain" at line 192

Another issue appear :

Error: "Call to undefined method FactionsPro\FactionMain::getFactionPower()" (EXCEPTION) in "FactionsPro_1.4_dev-26.phar/src/FactionsPro/FactionMain" at line 198

FactionsPro_1.4_dev-26.phar
PocketMine-MP 1.7dev-379

```2017-11-07 [18:28:21] [Server thread/CRITICAL]: Unhandled exception executing command 'f claim' in f: Call to undefined method FactionsPro\FactionMain::getFactionPower()
2017-11-07 [18:28:21] [Server thread/CRITICAL]: Error: "Call to undefined method FactionsPro\FactionMain::getFactionPower()" (EXCEPTION) in "FactionsPro_1.4_dev-26.phar/src/FactionsPro/FactionMain" at line 198
2017-11-07 [18:28:21] [Server thread/DEBUG]: #0 FactionsPro_1.4_dev-26.phar/src/FactionsPro/FactionCommands(340): FactionsPro\FactionMain->drawPlot(pocketmine\Player object, string Apoc, double 5403, double 71, double 510, pocketmine\level\Level object, integer 25)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #1 FactionsPro_1.4_dev-26.phar/src/FactionsPro/FactionMain(94): FactionsPro\FactionCommands->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2017-11-07 [18:28:21] [Server thread/DEBUG]: #2 src/pocketmine/command/PluginCommand(58): FactionsPro\FactionMain->onCommand(pocketmine\Player object, pocketmine\command\PluginCommand object, string f, array Array())
2017-11-07 [18:28:21] [Server thread/DEBUG]: #3 src/pocketmine/command/SimpleCommandMap(258): pocketmine\command\PluginCommand->execute(pocketmine\Player object, string f, array Array())
2017-11-07 [18:28:21] [Server thread/DEBUG]: #4 src/pocketmine/Server(1979): pocketmine\command\SimpleCommandMap->dispatch(pocketmine\Player object, string f claim)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #5 src/pocketmine/Player(2168): pocketmine\Server->dispatchCommand(pocketmine\Player object, string f claim)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #6 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(220): pocketmine\Player->chat(string /f claim)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #7 src/pocketmine/network/mcpe/protocol/CommandRequestPacket(47): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleCommandRequest(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #8 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\CommandRequestPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #9 src/pocketmine/network/mcpe/protocol/BatchPacket(115): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\CommandRequestPacket object)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #10 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(92): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #11 src/pocketmine/Player(3052): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #12 src/pocketmine/network/mcpe/RakLibInterface(143): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #13 src/raklib/server/ServerHandler(99): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 82.124.37.233:41063, raklib\protocol\EncapsulatedPacket object, integer 0)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #14 src/pocketmine/network/mcpe/RakLibInterface(84): raklib\server\ServerHandler->handlePacket()
2017-11-07 [18:28:21] [Server thread/DEBUG]: #15 src/pocketmine/network/Network(89): pocketmine\network\mcpe\RakLibInterface->process()
2017-11-07 [18:28:21] [Server thread/DEBUG]: #16 src/pocketmine/Server(2513): pocketmine\network\Network->processInterfaces()
2017-11-07 [18:28:21] [Server thread/DEBUG]: #17 src/pocketmine/Server(2269): pocketmine\Server->tick()
2017-11-07 [18:28:21] [Server thread/DEBUG]: #18 src/pocketmine/Server(2144): pocketmine\Server->tickProcessor()
2017-11-07 [18:28:21] [Server thread/DEBUG]: #19 src/pocketmine/Server(1734): pocketmine\Server->start()
2017-11-07 [18:28:21] [Server thread/DEBUG]: #20 src/pocketmine/PocketMine(564): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/, string /home/gsp_1485/g3.hmserv.eu-19144/, string /home/gsp_1485/g3.hmserv.eu-19144/plugins/)
2017-11-07 [18:28:21] [Server thread/DEBUG]: #21 /home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar(1): require(string phar:///home/gsp_1485/g3.hmserv.eu-19144/PocketMine-MP.phar/src/pocketmine/PocketMine.php)```

**Comment by Awzaw** [comment]
This confirms what I expected that FactionsBetaPro is a very incomplete rewrite of FactionsPro :-/ This is fixed in 2185b2a, and the new latest Poggit release

**Comment by teracube** [comment]
@Awzaw 
i tested 2185b2a on my server and it's ok
thanks
